fre/make/gfdlfremake/yamlfre.py

```diff
@@ -53,14 +53,17 @@ def __init__(self,compileinfo):
         """
         # compile information from the combined yaml
         self.yaml = compileinfo
+        # Check if self.yaml is None
+        if self.yaml is None:
+            raise ValueError("The provided compileinfo is None. It must be a valid dictionary.")
 
-        ## Check the yaml for required things
         ## Check for required experiment name
         try:
             self.yaml["experiment"]
-        except:
+        except KeyError:
             print("You must set an experiment name to compile \n")
             raise
```

Comment:
Instead of making these separate, you can make this `raise KeyError("You must set an experiment name to compile \n")` - similar to line 58